src/bsp/Bsp.cpp

```diff
@@ -2802,6 +2886,19 @@ int Bsp::add_texture(const char* name, byte* data, int width, int height) {
 		}
 	}
 
+	if (oldtex != nullptr)
+	{
+		memcpy((byte*)oldtex + oldtex->nOffsets[0], mip[0], width * height);
+		memcpy((byte*)oldtex + oldtex->nOffsets[1], mip[1], (width >> 1) * (height >> 1));
+		memcpy((byte*)oldtex + oldtex->nOffsets[2], mip[2], (width >> 2) * (height >> 2));
+		memcpy((byte*)oldtex + oldtex->nOffsets[3], mip[3], (width >> 3) * (height >> 3));
+		memcpy((byte*)oldtex + (oldtex->nOffsets[3] + (width >> 3) * (height >> 3) + 2), palette, sizeof(COLOR3) * 256);
+		for (int i = 0; i < MIPLEVELS; i++) {
+			delete[] mip[i];
+		}
+		return 0;
+	}
+

```

Comment:
the new texture might be a different size, which could cause a crash or corruption.

Reply:
But it fixed now at start function

Reply:
oldtex = null if no matches

Reply:
Oh I see that now. No problem then. 👍